kiwi/firmware.py

```diff
@@ -43,7 +43,7 @@ def __init__(self, xml_state):
         ]
 
         firmware_types = Defaults.get_firmware_types()
-        if self.host_architecture.startswith('ppc64'):
+        if not self.firmware and self.host_architecture.startswith('ppc64'):
             self.firmware = 'ofw'
```

Comment:
I agree so let us create a map in Defaults which defines a default firmware for each supported arch, it could also be None. I was thinking like

```
diff --git a/kiwi/defaults.py b/kiwi/defaults.py
index e448a6d..9a868d8 100644
--- a/kiwi/defaults.py
+++ b/kiwi/defaults.py
@@ -223,6 +223,27 @@ class Defaults(object):
         }
 
     @classmethod
+    def get_default_firmware(self, arch):
+        default_firmware = {
+            'x86_64': 'bios',
+            'i586': 'bios',
+            'i686': 'bios',
+            'aarch64': None,
+            'arm64': None,
+            'armv5el': None,
+            'armv5tel': None,
+            'armv6l': None,
+            'armv7l': None,
+            'ppc': 'ofw',
+            'ppc64': 'ofw',
+            'ppc64le': 'ofw',
+            's390': None,
+            's390x': None
+        }
+        if arch in default_firmware:
+            return default_firmware(arch)
+
+    @classmethod
     def get_default_boot_timeout_seconds(self):
         return 10
```

and in firmware we just call

```
if not self.firmware:
    self.firmware = Defaults.get_default_firmware(self.host_architecture)
```

Thoughts ?